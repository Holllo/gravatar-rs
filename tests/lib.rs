use gravatar_rs::Generator;

const BAUKE_EMAIL: &str = "me@bauke.xyz";
const HOLLLO_EMAIL: &str = "helllo@holllo.cc";

#[test]
fn test_hash_email() {
  let samples = [("bauke", BAUKE_EMAIL), ("holllo", HOLLLO_EMAIL)];

  for (name, email) in samples {
    insta::assert_snapshot!(
      format!("hash-{name}"),
      Generator::hash_email(email)
    );

    insta::assert_snapshot!(
      format!("hash-{name}-whitespace"),
      Generator::hash_email(&format!("  {email}  "))
    );

    insta::assert_snapshot!(
      format!("hash-{name}-casing"),
      Generator::hash_email(&email.to_uppercase())
    );
  }
}

#[test]
fn test_generator() {
  let emails = [BAUKE_EMAIL, HOLLLO_EMAIL];
  let samples = [
    ("gravatar", Generator::default().base_url),
    ("libravatar", "cdn.libravatar.org".to_string()),
  ];

  for (name, base_url) in samples {
    let generator = Generator::default().set_base_url(&base_url);
    let urls = emails.map(|email| generator.generate(email));
    insta::assert_debug_snapshot!(format!("generate-{name}"), urls);
  }
}
