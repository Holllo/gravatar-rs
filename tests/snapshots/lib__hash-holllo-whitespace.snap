---
source: tests/lib.rs
expression: "Generator::hash_email(&format!(\"  {email}  \"))"
---
ebff9105dce4954b1bdb57fdab079ff3
